moodlecli/main.py

```diff
@@ -250,3 +250,33 @@ def export_users_by_course(source_course_id, bucket_name, key):
     )
 
     aws.put_json_data(users, bucket_name, key)
+
+
+@cli.command()
+@click.argument('bucket_name')
+@click.argument('directory')
+def export_users_bulk(bucket_name, directory):
+    "Output the users for a series of classes to an s3 bucket"
+    moodle = get_moodle_client()
+
+    course_data = moodle.get_courses()
+    course_ids = utils.get_course_ids(course_data)
+    for id in course_ids:
+        user_data = moodle.get_users_by_course(id)
+        key = f'{directory}/users_course-{id}.json'
+        aws.put_json_data(user_data, bucket_name, key)
+
+
+@cli.command()
+@click.argument('bucket_name')
+@click.argument('directory')
+def export_grades_bulk(bucket_name, directory):
```

Comment:
:thinking: - IDK if this would be a better CLI experience, but what do you think about having a single `export-bulk` command which takes a required [choice option](https://click.palletsprojects.com/en/8.0.x/options/#choice-options) (e.g. `moodle-cli export-bulk bucket directory --data=grades). I'm not convinced myself if I like that better but wanted to throw the idea out.

Reply:
This could be interesting - maybe a little less user friendly, but would clean up the command menu. In the end, it's only the sys admins that will be using the CLI. If we implement it so that a csv directs which classes to export then it would be a nice, extensible way to get more data out of a series of classes if we need it... i'm thinking time spent, manual course content updates, or hidden materials...

Reply:
Yeah, agreed. I'll leave it up to your judgement as to whether to have two separate commands or try to consolidate into one with an option flag. I can see pros and cons for both :man_shrugging: .